.github/workflows/vapoursynth.yml

```diff
@@ -0,0 +1,151 @@
+
+jobs:
+  clippy-rustfmt:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - name: Install stable
+      uses: actions-rs/toolchain@v1
+      with:
+        profile: minimal
+        toolchain: stable
+        override: true
+        components: clippy, rustfmt
+    - name: Run clippy
+      uses: actions-rs/clippy-check@v1
+      with:
+        token: ${{ secrets.GITHUB_TOKEN }}
+        args: --all --all-features
+    - name: Run rustfmt
+      uses: actions-rs/cargo@v1
+      with:
+        command: fmt
+        args: --all -- --check
+
+  unix-tests:
+    strategy:
+      matrix:
+        toolchain:
+         - stable-x86_64-apple-darwin
+         - stable-x86_64-unknown-linux-gnu
+         - stable-i686-unknown-linux-gnu
+        include:
+         - toolchain: stable-x86_64-apple-darwin
+           os: macOS-latest
+         - toolchain: stable-x86_64-unknown-linux-gnu
+           os: ubuntu-latest
+         - toolchain: stable-i686-unknown-linux-gnu
+           os: ubuntu-latest
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+    - uses: actions/checkout@v1
+    - name: Install VapourSynth on MacOS
+      if: matrix.os == 'macOS-latest'
+      run: |
+        brew install vapoursynth
+    - name: Install VapourSynth on Ubuntu 64-bit
+      if: matrix.toolchain == 'stable-x86_64-unknown-linux-gnu'
+      run: |
+        build/script.sh
+    - name: Install VapourSynth on Ubuntu 32-bit
+      if: matrix.toolchain == 'stable-i686-unknown-linux-gnu'
+      run: |
+        build/script.sh --build=i686-linux-gnu \
+                        "CFLAGS=-m32" "CXXFLAGS=-m32" "LDFLAGS=-m32"
+    - name: Install stable
+      uses: actions-rs/toolchain@v1
+      with:
+        profile: minimal
+        toolchain: ${{ matrix.toolchain }}
+        override: true
+    - name: Run tests
+      run: |
+        python3 build/run-tests.py
+    - name: Run sample-plugin
+      env:
+        FUNCS: vapoursynth/vapoursynth-functions vapoursynth/vsscript-functions
+      run: |
+        cd sample-plugin
+        cargo build --verbose
+        cargo run --verbose --bin test --features cfg-if $FUNCS
+    - name: Run doc
+      uses: actions-rs/cargo@v1
+      with:
+        command: doc
+        args: --all-features --verbose
+    - name: Copy index into the target directory
+      if: matrix.toolchain == 'stable-x86_64-unknown-linux-gnu'
+      run: |
+        cp build/index.html target/doc
+    - name: VapourSynth deployment
+      if: matrix.toolchain == 'stable-x86_64-unknown-linux-gnu'
+      uses: JamesIves/github-pages-deploy-action@master
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        BASE_BRANCH: master
+        BRANCH: gh-pages
+        FOLDER: target/doc
+
+  windows-tests:
+    strategy:
+      matrix:
+        toolchain:
+         - stable-x86_64-pc-windows-msvc
+         - stable-x86_64-pc-windows-gnu
+         - stable-i686-pc-windows-msvc
+         - stable-i686-pc-windows-gnu
```

Comment:
Would it be possible to add beta and nightly here too?

Reply:
Same as above